### PR TITLE
Don't call Delete if the CR has already been deleted

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Fixes https://github.com/tigera/operator/issues/1510

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by caseydavenport** [comment]
@tmjd what do you think about this one?

File: pkg/controller/status/status.go [hunk 1]
```diff
@@ -106,6 +106,7 @@ type statusManager struct {
 	// if there are any, and report statuses based on the state of those resources.
 	readyToMonitor bool
 	hasSynced      bool
+	crExists       bool
```
Comment (tmjd) [inline pkg/controller/status/status.go:109]: Should we default this to true to ensure if the operator is restarted that an old status would get cleaned up if needed?

Comment (caseydavenport) [reply pkg/controller/status/status.go:109]: Yeah that's probably a good idea.

Comment (tmjd) [reply pkg/controller/status/status.go:109]: Maybe a better idea would be to query for the CR in the New call so we initialize to the real value.

**Review by tmjd — APPROVED** [review]
LGTM, don't forget to squash the commits.

**Pull request merged**